### PR TITLE
Redesign attachments display

### PR DESCRIPTION
**Pull request by wrightmartin**
Part of #3269 

This adds a redesigned attachments section to correspondence

(I can't add a screenshot atm due to poor wifi, but will shortly)

- Improve the handling of long lists of attachments by hiding lists longer than 3, with a 'reveal more' button
- Adds the HTML for a 'download all' attachments button (needs implementing)

todo:
- [ ] ~~Download all functionality~~
- [ ] Attachments action button (link to attachments bookmark, quick 'download all' link), depends on implementing action buttons

### CODE REVIEW

File: app/views/request/_bubble.html.erb [hunk 1]
```diff
@@ -42,8 +42,30 @@
...
+    <a href="#" title="" id="js-attachments__show-more" class="attachments__show-more"><%= _('View all attachments') %> (<span class="js-attachment-count"></span>)</a>
   </div>
   <% end %>
 
   <p><%= body %></p>
+  <script>
+    var limit = 3;
+    var per_page = 1000; //change this to enable 'pages' (reveal per_page at a time)
+
+    jQuery('.list-of-attachments > .attachment:gt('+(limit-1)+')').hide();
```
Comment (wrightmartin) [inline app/views/request/_bubble.html.erb]: Should this be moved to general.js?

**Comment by wrightmartin** [comment]
Part of #3269

File: app/views/request/_bubble.html.erb (hunk 1 above)
Comment (garethrees) [inline app/views/request/_bubble.html.erb]: Yeah, that would make sense

**Comment by garethrees** [comment]
Dropping 8e4742c because we don't have the backend code to be able to make this happen yet.

**Comment by garethrees** [comment]
wheeeee!

attachments-expand-contract

**Pull request merged**